Documentation/internals/security-identities.rst

```diff
@@ -0,0 +1,62 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+*******************
+Security Identities
+*******************
+
+Security identities are generated from labels. They are stored as ``uint32``,
+which means the maximum limit for a security identity is ``2^32 - 1``. The
+minimum security identity is ``1``.
+
+.. note::
+
+   Identity 0 is not a valid value. If it shows up in Hubble output, this means
+   the identity was not found. In the eBPF datapath, it has a special role
+   where it denotes "any identity", i.e. as a wildcard allow in policy maps.
+
+Security identities span over several ranges, depending on the context:
+
+1) Cluster-local
+2) ClusterMesh
+3) Identities generated from CIDR-based policies
+
+Cluster-local identities (1) range from ``1`` to ``2^16 - 1``. The lowest
+values, from ``1`` to ``255``, correspond to the reserved identity range.  See
+the `internal code documentation
+<https://pkg.go.dev/github.com/cilium/cilium/pkg/identity#NumericIdentity>`__
+for details.
+
+For ClusterMesh (2), 8 bits are used as the ``cluster-id`` which identifies the
+cluster in the ClusterMesh, into the 3rd octet as shown by ``0x00FF0000``. This
+does not apply to CIDR identities however, see (3).
+
+CIDR identities (3) are local to each node. CIDR identities begin from ``1``
+and end at ``16777215``, however since they're shifted by ``24``, this makes
+their effective range ``1 | (1 << 24)`` to ``16777215 | (1 << 24)`` or from
+``16777217`` to ``33554431``.
+
+When CIDR policies are applied, the identity generated is local to each node.
+In other words, the identity may not be the same for the same CIDR policy
+across two nodes. Additionally, CIDR identities are not included within the
+numerical identities range ``256`` to ``2^24 - 1``), as they're stored
+separately inside the Agent.
```

Comment:
I'm a little bit confused what this last sentence means. The previous paragraph already defined that the range starts at 2^24. What does it mean for the identities to be stored separately inside the agent?

Reply:
:+1: I've removed this sentence, as it's not adding value. Probably a remnant after all the changes. I also merged it with the previous paragraph.